apps/src/componentLibrary/dialog/Dialog.tsx

```diff
@@ -1,23 +1,36 @@
 import classnames from 'classnames';
-import React, {HTMLAttributes, ReactNode} from 'react';
+import React, {HTMLAttributes, ReactNode, useRef} from 'react';
 
-import {Button} from '@cdo/apps/componentLibrary/button';
+import {Button, ButtonProps} from '@cdo/apps/componentLibrary/button';
 import CloseButton from '@cdo/apps/componentLibrary/closeButton';
+import useBodyScrollLock from '@cdo/apps/componentLibrary/common/hooks/useBodyScrollLock';
+import useEscapeKeyHandler from '@cdo/apps/componentLibrary/common/hooks/useEscapeKeyHandler';
+import useFocusTrap from '@cdo/apps/componentLibrary/common/hooks/useFocusTrap';
 import FontAwesomeV6Icon, {
   FontAwesomeV6IconProps,
 } from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import {BodyTwoText, Heading2} from '@cdo/apps/componentLibrary/typography';
 
 import moduleStyles from './dialog.module.scss';
 
 export interface DialogProps extends HTMLAttributes<HTMLDivElement> {
-  /** Dialog type*/
-  type?: 'noIcon' | 'withIconFill' | 'withIconFA';
   /** Dialog title */
   title?: string;
-  /** Dialog content */
-  content?: string | ReactNode;
-  /** Whether to show secondary button */
-  showSecondaryButton?: boolean;
+  /** Dialog description text */
+  description?: string;
+  /** Dialog Custom content (rendered right after/instead Dialog description) */
+  customContent?: ReactNode;
+  /** Custom bottom content (rendered right after Dialog actions section).
+   *  If this is rendered when there's no `description` prop - make sure to add `dsco-dialog-description` `id`
```

Comment:
Is there a way we can enforce this?

Reply:
Good question!
I think we can do smth like following:

```
  useEffect(() => {
    if (!description && customContent) {
      const hasDescriptionId = dialogRef.current?.querySelector(
        '#dsco-dialog-description'
      );
      if (!hasDescriptionId) {
        console.warn(
          "Warning: When 'description' is not provided, customContent must include an element with ID 'dsco-dialog-description' for accessibility."
        );
      }
    }
  }, [description, customContent]);
```

What do you think?

Reply:
I updated the comment cause it should have been placed nex to customContent, not customButtonContent.

As for enforce the error - I suggest we update in a follow up PR if needed

Reply:
Sounds good, yeah a follow-up is fine